src/index.ts

```diff
@@ -73,42 +76,28 @@ const validateSpecificationFolderMap = async (cwd: string) => {
   return map
 }
 
-const sourceBranch = "source-b6791c5f-e0a5-49b1-9175-d7fd3e341cb8"
-
 /**
  * The function validates files in the given `cwd` folder and returns errors.
  *
- * @param { cwd, env }
+ * @param config
  */
-export const avocado = ({ cwd, env }: cli.Config): asyncIt.AsyncIterableEx<Error> =>
+export const avocado = (config: cli.Config): asyncIt.AsyncIterableEx<Error> =>
   asyncIt.iterable<Error>(async function*() {
-    const targetBranch = env.SYSTEM_PULLREQUEST_TARGETBRANCH
+    const pr = await azureDevOps.createPullRequestProperties(config)
     // detect Azure DevOps Pull Request validation.
-    if (targetBranch !== undefined) {
-      const sourceGitRepository = git.repository(cwd)
-      await sourceGitRepository({ branch: [sourceBranch] })
-      await sourceGitRepository({ branch: [targetBranch, `remotes/origin/${targetBranch}`] })
-
-      // we have to clone the repository because we need to switch branches.
-      // Switching branches in the current repository can be dangerous because Avocado
-      // may be running from it.
-      const target = path.resolve(path.join(cwd, "..", "target"))
-      await fs.mkdir(target)
-      const targetGitRepository = git.repository(target)
-      await targetGitRepository({ clone: [cwd, "."] })
-
-      await targetGitRepository({ checkout: [targetBranch] })
-      const targetMap = await validateSpecificationFolderMap(target)
+    if (pr !== undefined) {
+      await pr.checkout(pr.targetBranch)
+      const targetMap = await validateSpecificationFolderMap(pr.workingDir)
 
-      await targetGitRepository({ checkout: [sourceBranch] })
-      const sourceMap = await validateSpecificationFolderMap(target)
+      await pr.checkout(pr.sourceBranch)
+      const sourceMap = await validateSpecificationFolderMap(pr.workingDir)
```

Comment:
seems like the validation logic only needs pr.workingDir, I'd suggest to refactor this "avocado" function. Have "avocado" take in the working dir as the main parameter, and have one maybe called "avocadoForAzureDevOps" which takes anything required to access Azure Dev Ops and obtain what you need to pass to "avocado". This way if we want to plug it in somewhere else, or just call it from the commandline we could just call "avocado" with the workingDir (string) parameter.

Reply:
also, please add a comment when you make the cal to the function, on what you're intending to validate so it's easier to read what validation is happening, and at the function definition too.